jobs/JGLOBAL_STAGE_IC

```diff
@@ -3,15 +3,38 @@
 source "${HOMEgfs}/ush/preamble.sh"
 source "${HOMEgfs}/ush/jjob_header.sh" -e "stage_ic" -c "base stage_ic"
 
-# Restart conditions for GFS cycle come from GDAS
+# Define significant cycles
 # shellcheck disable=SC2153
-rCDUMP=${RUN}
-# shellcheck disable=SC2153
-[[ ${RUN} = "gfs" ]] && export rCDUMP="gdas"
-export rCDUMP
+half_window=$(( assim_freq / 2 ))
+current_cycle="${PDY}${cyc}"
+previous_cycle=$(date --utc -d "${PDY} ${cyc} - ${assim_freq} hours" +%Y%m%d%H)
+current_cycle_begin=$(date --utc -d "${PDY} ${cyc} - ${half_window} hours" +%Y%m%d%H)
+current_cycle_end=$(date --utc -d "${PDY} ${cyc} + ${half_window} hours" +%Y%m%d%H)
+
+# Define model start date for current_cycle as the time the forecast will start
+if [[ "${DOIAU:-NO}" == "YES" ]] && [[ "${MODE}" == "cycled" ]] ; then
+  model_start_date_current_cycle="${current_cycle_begin}"
+else
+  if [[ "${REPLAY_ICS:-NO}" == "YES" ]]; then
+    model_start_date_current_cycle=${current_cycle_end}
+  else
+    model_start_date_current_cycle=${current_cycle}
+  fi
+fi
+export current_cycle previous_cycle model_start_date_current_cycle
+
+# Initialize return code
+err=0
+
+# Execute staging
+"${SCRgfs}/exglobal_stage_ic.py"
 
-# Execute the Script
-"${SCRgfs}/exglobal_stage_ic.sh"
+###############################################################
+# Check for errors and exit if any of the above failed
+if [[ "${err}" -ne 0 ]]; then
```

Comment:
When will this be not 0?  It is set to 0 on line 27 and never reset?

Reply:
You are correct, have added missing error value check after existing script. Committed to branch @ 7ae9ea49